pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/medicalrecord/MedicalRecordInstanceProvider.java

```diff
@@ -19,7 +19,8 @@ public class MedicalRecordInstanceProvider implements DDPInstanceProvider {
      */
     @Override
     public List<DDPInstance> getApplicableInstances() {
-        return DDPInstance.getDDPInstanceListWithRole(DBConstants.HAS_MEDICAL_RECORD_ENDPOINTS);
+        return DDPInstance.getDDPInstanceListWithRole(DBConstants.HAS_MEDICAL_RECORD_ENDPOINTS).stream()
```

Comment:
I think we need a method comment on getDDPInstanceListWithRole, and or a javadoc comment on getApplicableInstances().  I read getDDPInstanceListWithRole and think "this returns ddp instances that have the given role".  But that appears to not be the case, since this additional filter must be applied.

Reply:
It looks like other uses of getDDPInstanceListWithRole() (outside of test code) do this addition hasRole() on the client side, so presumably they are doing the "right" thing.

Reply:
Added a comment.
Yes, I agree. There is no good reason this method should not return a filtered list.